tests/ops/op_math/test_pow_op.py

```diff
@@ -317,6 +318,26 @@ def test_has_matrix_false(self, power_method):
 
         assert op.has_matrix is False
 
+    @pytest.mark.parametrize("z", [-2, 3, 2])
+    def test_has_adjoint_true(self, z, power_method):
+        """Test `has_adjoint` property is true for integer powers."""
+        # Note that even if the base would have `base.has_adjoint=False`, `qml.adjoint`
+        # would succeed because it would create an `Adjoint(base)` operator.
+        base = qml.PauliX(0)
+        op: Pow = power_method(base=base, z=z)
+
+        assert op.has_adjoint is True
+
+    @pytest.mark.parametrize("z", [-2.0, 1.0, 0.32])
```

Comment:
And the first two values would need to be changed, right? 
```suggestion
    @pytest.mark.parametrize("z", [-0.5, 0.32])
```

Based on this test, it seems like you had already considered whole-floats but deemed them as a fraction on purpose. I am open to be corrected :o